rfcs/0221-change-since-relative-timestamp/0221-change-since-relative-timestamp.md

```diff
@@ -0,0 +1,76 @@
+---
+Number: "0221"
+Category: Consensus (Soft Fork)
+Status: Draft
+Author: Ian Yang
+Organization: Nervos Foundation
+Created: 2021-02-03
+---
+
+# Use input cell committing block timestamp as the start time for the relative timestamp in `since`
+
+## Abstract
+
+This document proposes a transaction verification consensus change. When the `since` field of the transaction input uses a relative timestamp, the referenced cell committing block timestamp is used instead of the median timestamp.
+
+This is a modification to the RFC17 [Transaction valid since](../0017-tx-valid-since/0017-tx-valid-since.md).
+
+## Motivation
+
+The current consensus rule uses the median of the timestamp fields of the 37 blocks preceding the referenced cell committing block. It is resource consuming to get the median timestamp because it requires either getting 37 block headers or caching the median timestamp for each block. The intention to use the median time is to avoid miners profitable by manipulating block timestamp to include more transactions, but it is safe to use the committing block timestamp as the start time because of two reasons:
+
+1. The timestamp in the block header has already been verified by the network that it must be larger than the median of the previous 37 blocks and less than or equal to the current time plus 15 seconds. TODO: reference to block data structure RFC
+2. The transaction consuming a cell with the `since` requirement is committed later than the transaction creating the cell. When miners see the consuming transaction, the start timestamp has already been determined. 
+
+## Specification
+
+When an input `since` field is present, and
+
+* The `metric_flag` is block timestamp (10).
+* The `relative_flag` is relative (1).
+
+The transaction is mature when
+
+```
+MedianTimestamp ≥ StartTime + SinceValue
+```
+
+where
+
+* `StartTime` is the block timestamp that committed the cell consumed by the input.
+* `SinceValue` is the `value` part of the `since` field.
+* `MedianTimestamp` is the median timestamp of the previous 37 blocks preceding the block if the transaction is in the block, or the latest 37 blocks if the transaction is in the pool.
+
+The transaction verification fails if the transaction is immature.
+
+The only change is `StartTime`, which was the median of the previous 37 blocks preceding the one that has committed the consumed cell. Because block timestamp must be larger than the median of its previous 37 blocks, the new consensus rule is stricker than the old rule. The transaction that is mature using the old rule may be immature using the new rule, but the transaction that is mature using the new rule must be mature using the old rule.
```

Comment:
```suggestion
The only change is `StartTime`, which was the median of the previous 37 blocks preceding the one that has committed the consumed cell. Because block timestamp must be larger than the median of its previous 37 blocks, the new consensus rule is stricter than the old rule. The transaction that is mature using the old rule may be immature using the new rule, but the transaction that is mature using the new rule must be mature using the old rule.
```